### PR TITLE
Added support for minidumps generating

### PR DESCRIPTION
**Pull request by SNMetamorph**
Solves #1205

### CODE REVIEW

File: engine/common/crashhandler.c [hunk 1]
```diff
@@ -189,6 +190,62 @@ static void Sys_StackTrace( PEXCEPTION_POINTERS pInfo )
...
+
+static void Sys_GetMinidumpFileName( const char *processName, char *mdmpFileName, size_t bufferSize )
+{
+	time_t currentUtcTime = time( NULL );
+	struct tm *currentLocalTime = localtime( &currentUtcTime );
+
+	Q_snprintf( mdmpFileName, bufferSize, "%s_%s_crash_%d%.2d%.2d_%.2d%.2d%.2d.mdmp",
+		processName,
+		Q_buildcommit(),
+		currentLocalTime->tm_year + 1900,
```
Comment (a1batross) [inline engine/common/crashhandler.c:208]: You can use strftime for formatting the date

Comment (a1batross) [reply engine/common/crashhandler.c:208]: Nevermind, it's fine too.

**Comment by a1batross** [comment]
Good job!

**Comment by a1batross** [comment]
I will test it on my machine and merge. :)

**Comment by a1batross** [comment]
It worked for me under Wine. Though MiniDumps here aren't so mini, got 400 MB file, that got compressed to 100 MB with Zstd. While it was generating, engine got freezed for few seconds, so that could be a problem. Would be nice to print them in Sys_Crash but that could be solved later.

Merging.

**Comment by SNMetamorph** [comment]
> While it was generating, engine got freezed for few seconds, so that could be a problem. Would be nice to print them in Sys_Crash but that could be solved later.
> 
> Merging.

May be because of Wine? On my setup it's writes almost immediately. And about compression: LZMA does very good with minidimps.

**Pull request merged**